lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -419,6 +419,28 @@ class DecomposeAtenAddmmOp : public OpRewritePattern<AtenAddmmOp> {
 };
 } // namespace
 
+// Decompose torch.mean into: sum(x)/div(numTensorElements).
+namespace {
+class DecomposeAtenMeanOp : public OpRewritePattern<AtenMeanOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenMeanOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value input = op.self();
+    Value output = op.result();
+    BaseTensorType inputTensorType = input.getType().cast<BaseTensorType>();
+    BaseTensorType outputTensorType = output.getType().cast<BaseTensorType>();
+    Value dtype = rewriter.create<ConstantNoneOp>(loc);
+    Value sum = rewriter.create<AtenSumOp>(loc, outputTensorType, input, dtype);
```

Comment:
At the moment the lowering of `AtenSumOp` does not use the `dtype` arg, so setting it to `None` does not cause issues, but I think it would be better if we did not assume that about `AtenSumOp`. The correct expectation for the inputs to `AtenSumOp` is that if the `outputTensorType` is different from the `input` type, then `outputTensorType` and `dtype` must match. You should be able to do this by simply changing `dtype` to `op.dtype()`, and `RefineTypes` will make sure that if `dtype` is not `None`, it matches `outputTensorType`.

It would also be nice to have a test for the `dtype` flag. See 

https://github.com/llvm/torch-mlir/blob/c965f9f90f544de14fa0ccf216ace3616891da79/e2e_testing/torchscript/reduction.py#L33

for an example.